src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -303,8 +308,13 @@ private void OutputFlags(NodeFactory factory, ref ObjectDataBuilder objData)
         {
             UInt16 flags = EETypeBuilderHelpers.ComputeFlags(_type);
 
+            if (_type.GetTypeDefinition() == factory.ArrayOfTEnumeratorType)
+            {
+                // Generic array enumerators use special variance rules recognized by the runtime
+                flags |= (UInt16)EETypeFlags.GenericVarianceFlag;
```

Comment:
I don't particularly like this special casing around ArrayOfTEnumeratorType (neither of them, actually). Might be worth spending some time to clean it up.